python/cuspatial/cuspatial/core/_column/geocolumn.py

```diff
@@ -30,23 +31,118 @@ def __init__(
         data: Tuple,
         meta: GeoMeta = None,
         shuffle_order: cudf.Index = None,
+        from_read_polygon_shapefile=False,
     ):
-        if (
-            not isinstance(data[0], cudf.Series)
-            or not isinstance(data[1], cudf.Series)
-            or not isinstance(data[2], cudf.Series)
-            or not isinstance(data[3], cudf.Series)
+        if from_read_polygon_shapefile:
+            """
+            A cudf.ListSeries needs four levels of nesting to represent
+            a polygon shapefile. The rings_offsets and polygons_offsets
+            have already been computed in the `read_polygon_shapefile`
+            function. In order to convert it into an arrow list<...>
+            we need a set of offsets buffers for each point tuple, and
+            and also an offsets buffer for each individual point.
+
+            Coordinates: List of size 2 offsets
```

Comment:
What is a "size 2 offset"?

Reply:
I changed it to "length 2" to match the below line of length 1 offsets. I also added a small example.